source/adapters/cuda/image.cpp

```diff
@@ -1214,6 +1241,9 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesWaitExternalSemaphoreExp(
     enqueueEventsWait(hQueue, Stream, numEventsInWaitList, phEventWaitList);
 
     CUDA_EXTERNAL_SEMAPHORE_WAIT_PARAMS SemWaitParams = {};
+    if (hasValue) {
+      SemWaitParams.params.fence.value = waitValue;
```

Comment:
If you don't set this `value` parameter then do you know if there a default value for it in constructor of `CUDA_EXTERNAL_SEMAPHORE_WAIT_PARAMS`?
There are loads of other params that this struct takes : https://docs.nvidia.com/cuda/archive/11.6.2/cuda-driver-api/structCUDA__EXTERNAL__SEMAPHORE__WAIT__PARAMS__v1.html#structCUDA__EXTERNAL__SEMAPHORE__WAIT__PARAMS__v1_188ca51398134e22437eedca531fcb420

Don't you have to set any of them?

Reply:
The `SemWaitParams` struct is constructed with ` = {}`, so all fields are zero initially. Currently we only support `CU_EXTERNAL_SEMAPHORE_HANDLE_TYPE_OPAQUE_FD`, for which no values need to be set in this struct according to the documentation, and `CU_EXTERNAL_SEMAPHORE_HANDLE_TYPE_D3D12_FENCE`, for which we need to set the `params.fence.value` field as we do here.

Any other fields are for use by semaphore types we do not currently support (e.g. keyed mutex), so we don't set any of them.

Reply:
Great, thanks for the clarifications.